src/frame.h

```diff
@@ -69,6 +70,10 @@ struct frame_additional_data : frame_header
 
     uint32_t raw_size = 0;     // The frame transmitted size (payload only)
 
+    // Used by software-device, to save metadata that doesn't have enough room in metadata_blob
```

Comment:
Can't we put it alongside the frame data, then? My concern is the overhead for every frame... there should be a way to do this only for the frames that need it, no?

Reply:
I also want to consider changing the previous `metadata_blob` from an `array` -- why not make that a vector (also removing `metadata_size`) or `map`?

Maybe set a meeting?

Reply:
The extra_data is used only by frames that need it.
The problem is that `software_device` uses `md_contant_parser` that saves the metadata as <enum, value> pairs, each the size of 12 bytes. That means that only 21 metadata items can fit in the 256 bytes `metadata_blob`. Current implementation was to throw all other metadata fields.
My implementation uses this vector to save all the fields that were previously discarded (and only them).

The drawback is that frames for other devices will need a bit more memory, but that seemed OK with all the temporary and other unused fields in the `additional_data` struct.

We considered several alternatives, and tested the performance of this implementation vs. keeping a database of metadata in the `dds_sensor_proxy`. This implementation was more efficient.

Reply:
I think we should change to a map and avoid complexity